### DISCUSSION TITLE
url helpers in tests not found

### DISCUSSION THREAD

**Discussion by megatux** [yippee-fun/phlex#683] [General]
Hi, I have errors running the autogenerated tests for my controller. The url helpers methods are not found.

I created a new rails app this way and ran tests (some edits & other installed gems are omitted):
```
rails new sample-bun_tailwind_phlex_htmx -j bun -c tailwind -d sqlite3 --skip-hotwire --skip-jbuilder --skip-bootsnap`

bundle add phlex-rails
bin/rails generate phlex:install
bin/rails g phlex:controller Articles index show

# running tests:

bin/rails test
....


# Running:

E

Error:
ArticlesControllerTest#test_should_get_show:
NameError: undefined local variable or method `articles_show_url' for an instance of ArticlesControllerTest
    test/controllers/articles_controller_test.rb:12:in `block in <class:ArticlesControllerTest>'


bin/rails test test/controllers/articles_controller_test.rb:11

E

Error:
ArticlesControllerTest#test_should_get_index:
NameError: undefined local variable or method `articles_index_url' for an instance of ArticlesControllerTest
    test/controllers/articles_controller_test.rb:7:in `block in <class:ArticlesControllerTest>'

```

**Comment by megatux** [comment]
Should I include `Phlex::Rails::Helpers::Routes` somewhere in the test?

**Comment by joeldrapper** [comment]
I can’t tell without seeing your test.

**Reply by megatux** [reply]
Oh, sure. I thought with the pasted commands and the output would be enough to duplicate it. There were no edits, just the generated code:

```ruby
require "test_helper"

class ArticlesControllerTest < ActionDispatch::IntegrationTest
  test "should get index" do
    get articles_index_url
    assert_response :success
  end

  test "should get show" do
    get articles_show_url
    assert_response :success
  end
end

``` 

The public repo for this app that uses Bun, Phlex, htmx and Tailwind [is here](https://github.com/megatux/rails_sample_bun_tailwind_phlex_htmx) (I fixed these tests in one of the latest commits).
Thanks!

**Reply by joeldrapper** [reply]
I think you might just be calling the wrong methods. I would expect this to be `articles_url` or `articles_path` and `article_url(article)` or `articles_url(article)` where `article` was an instance of a persisted `Article`. You can run `bin/rails routes -g article` to see what your paths are.

**Reply by joeldrapper** [reply]
I don’t understand how the test was generated like that, but we’re removing the controller generator anyway because none of us use it and no one wants to maintain it. 😅
